frontend/packages/dev-console/src/components/topology/Topology.tsx

```diff
@@ -53,23 +60,44 @@ const graphModel: Model = {
   },
 };
 
-const Topology: React.FC<TopologyProps> = ({ data, serviceBinding, filters, activeNamespace }) => {
+const Topology: React.FC<TopologyProps> = ({
+  data,
+  filters,
+  application,
+  namespace,
+  serviceBinding,
+}) => {
   const visRef = React.useRef<Visualization | null>(null);
+  const applicationRef = React.useRef<string>(null);
   const componentFactoryRef = React.useRef<ComponentFactory | null>(null);
+  const knativeComponentFactoryRef = React.useRef<KnativeComponentFactory | null>(null);
+  const helmComponentFactoryRef = React.useRef<HelmComponentFactory | null>(null);
+  const operatorsComponentFactoryRef = React.useRef<OperatorsComponentFactory | null>(null);
   const [layout, setLayout] = React.useState<string>(graphModel.graph.layout);
   const [model, setModel] = React.useState<Model>();
   const [graphData, setGraphData] = React.useState<GraphData>();
   const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
-  const createResourceAccess: string[] = useAddToProjectAccess(activeNamespace);
+  const createResourceAccess: string[] = useAddToProjectAccess(namespace);
 
   if (!componentFactoryRef.current) {
     componentFactoryRef.current = new ComponentFactory(serviceBinding);
   }
-
+  if (!knativeComponentFactoryRef.current) {
+    knativeComponentFactoryRef.current = new KnativeComponentFactory(serviceBinding);
+  }
+  if (!helmComponentFactoryRef.current) {
+    helmComponentFactoryRef.current = new HelmComponentFactory(serviceBinding);
+  }
+  if (!operatorsComponentFactoryRef.current) {
+    operatorsComponentFactoryRef.current = new OperatorsComponentFactory(serviceBinding);
+  }
```

Comment:
Curious. I think you can just do `React.useRef<Type>(new MyItem(value))`. I think that achieves the same net result.

Reply:
@jeff-phillips-18 any consideration to the above? It'll save you the use of the `if` statements.